treeherder/etl/seta.py

```diff
@@ -0,0 +1,210 @@
+    '''
+    build_system_type - buildbot or taskcluster
+    job_type_name - e.g. Mochitest (JobType.name)
+    platform_option - e.g. opt
+    ref_data_name - builder name
+    '''
+    # TODO: figure out how to ignore build, lint, etc. jobs
+    if build_system_type == 'buildbot':
+        return ref_data_name.split(' ')[-1]
+    else:
+        # e.g. "desktop-test-linux64/opt-reftest-8",
+        # taskcluster test types
+        testtype = job_type_name
+        if testtype.startswith('[funsize'):
+            return None
+
+        testtype = testtype.split('/opt-')[-1]
+        testtype = testtype.split('/debug-')[-1]
+
+        # this is plain-reftests for android
+        testtype = testtype.replace('plain-', '')
+
+        # SETA/Ouija do not care about builds, or other jobs like lint, etc.
+        # Bug 1318659 - for how mozci could help here
+        testtype = testtype.replace(' Opt', 'build')
+        testtype = testtype.replace(' Debug', 'build')
+        testtype = testtype.replace(' Dbg', 'build')
+        testtype = testtype.replace(' (opt)', 'build')
+        testtype = testtype.replace(' PGO Opt', 'build')
+        testtype = testtype.replace(' Valgrind Opt', 'build')
+        testtype = testtype.replace(' Artifact Opt', 'build')
+        testtype = testtype.replace(' (debug)', 'build')
+
+        testtype = testtype.strip(' ')
+
+        # TODO: these changes should have bugs on file to fix the names
+        testtype = testtype.replace('browser-chrome-e10s', 'e10s-browser-chrome')
+        testtype = testtype.replace('devtools-chrome-e10s', 'e10s-devtools-chrome')
+        testtype = testtype.replace('[TC] Android 4.3 API15+ ', '')
+
+        # TODO: fix this in updatedb.py
+        testtype = testtype.replace('webgl-', 'gl-')
+
+        return testtype
+
+
+def _get_group(name):
+    """It's about to get group for the test job, like crashtest for crashtest-1."""
+    try:
+        group = name.split('-')[0]
+    except:
+        group = ''
+    return group
+
+
+def _get_group_code(tbplnames, name):
+    """
+    It's about get group code by given testname, like T-e10s for tp5o-e10s.
+    And there has some useless blank in testname like " mochitest-devtools-chrome-3"
+    to strip that blank off, so we can get the code appropriately.
+    """
+    try:
+        code = tbplnames[name.strip()]['group']
+    except:
+        code = ''
+    return code
+
+
+def _get_code(tbplnames, name):
+    """
+    It's about get the code by given testname, it usually related with the group code
+    like '2' for web-platform-tests-2.   And there has some useless blank in testname
+    like " mochitest-devtools-chrome-3" to strip that blank off,
+    so we can get the code appropriately.
+    """
+    try:
+        code = tbplnames[name.strip()]['code']
+    except:
+        code = ''
+    return code
+
+
+def valid_platform(platform):
+    # TODO: This is very hardcoded and prone to fall out of date
```

Comment:
possibly a method for ensuring we do not get out of date here and in other hardcoded areas would be to have a weekly report of platforms we support, buildtypes, and unique names.  Ideally that report could help shed light on anything new from a previous week and would remind us to ensure that we are accounting for new names in SETA, TC configs, platform support, ownership, etc.